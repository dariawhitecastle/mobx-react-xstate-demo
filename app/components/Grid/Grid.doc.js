import { describe, PropTypes } from 'react-desc'

import { genericProps } from '../../utils/propTypes'
import { themeDocUtils } from '../../utils/themeDocUtils'

const fixedSizes = ['xsmall', 'small', 'medium', 'large', 'xlarge']
const sizes = [
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
  'full',
  '1/2',
  '1/3',
  '2/3',
  '1/4',
  '2/4',
  '3/4',
  'flex',
  'auto',
]
const edgeSizes = ['small', 'medium', 'large', 'none']

export const doc = Grid => {
  const DocumentedGrid = describe(Grid)
    .description(
      `A grid system for laying out content. To use, define the
rows and columns, create area names for adjacent cells, and then
place Box components inside those areas using the gridArea property.
See https://css-tricks.com/snippets/css/complete-guide-grid/.
The availability of Grid can be tested via \`Grid.available\`. Use this
to create fallback rendering for older browsers, like ie11.`,
    )
    .usage(
      `import { Grid } from 'components/Grid';
<Grid />`,
    )
    .intrinsicElement('div')

  DocumentedGrid.propTypes = {
    ...genericProps,
    align: PropTypes.oneOf(['start', 'center', 'end', 'stretch'])
      .description(
        `How to align the individual items inside the grid when there is extra
space in the column axis.`,
      )
      .defaultValue('stretch'),
    alignContent: PropTypes.oneOf([
      'start',
      'center',
      'end',
      'between',
      'around',
      'stretch',
    ]).description('How to align the contents along the column axis.'),
    areas: PropTypes.oneOf([
      PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          start: PropTypes.arrayOf(PropTypes.number),
          end: PropTypes.arrayOf(PropTypes.number),
        }),
      ),
      PropTypes.shape({
        xsmall: PropTypes.arrayOf(
          PropTypes.shape({
            name: PropTypes.string,
            start: PropTypes.arrayOf(PropTypes.number),
            end: PropTypes.arrayOf(PropTypes.number),
          }),
        ),
        small: PropTypes.arrayOf(
          PropTypes.shape({
            name: PropTypes.string,
            start: PropTypes.arrayOf(PropTypes.number),
            end: PropTypes.arrayOf(PropTypes.number),
          }),
        ),
        medium: PropTypes.arrayOf(
          PropTypes.shape({
            name: PropTypes.string,
            start: PropTypes.arrayOf(PropTypes.number),
            end: PropTypes.arrayOf(PropTypes.number),
          }),
        ),
        large: PropTypes.arrayOf(
          PropTypes.shape({
            name: PropTypes.string,
            start: PropTypes.arrayOf(PropTypes.number),
            end: PropTypes.arrayOf(PropTypes.number),
          }),
        ),
      }),
    ]).description('Area names and column,row coordinates.'),
    columns: PropTypes.oneOfType([
      PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.oneOf(sizes), PropTypes.string])),
          PropTypes.oneOf(sizes),
          PropTypes.string,
        ]),
      ),
      PropTypes.oneOf(fixedSizes),
      PropTypes.shape({
        count: PropTypes.oneOfType([PropTypes.oneOf(['fit', 'fill']), PropTypes.number]),
        size: PropTypes.oneOfType([
          PropTypes.oneOf(fixedSizes),
          PropTypes.arrayOf(PropTypes.oneOf(sizes)),
          PropTypes.string,
        ]),
      }),
      PropTypes.string,
    ]).description(
      `Column sizes.
      If an array value is an array, the inner array indicates the
      minimum and maximum sizes for the column.
      Specifying a single string will repeat multiple columns
      of that size, as long as there is room for more.
      Specifying an object allows indicating how the columns
      stretch to fit the available space.`,
    ),
    fill: PropTypes.oneOfType([
      PropTypes.oneOf(['horizontal', 'vertical']),
      PropTypes.bool,
    ]).description('Whether the width and/or height should fill the container.'),
    gap: PropTypes.oneOfType([
      PropTypes.oneOf(edgeSizes),
      PropTypes.shape({
        row: PropTypes.oneOfType([PropTypes.oneOf(edgeSizes), PropTypes.string]),
        column: PropTypes.oneOfType([PropTypes.oneOf(edgeSizes), PropTypes.string]),
      }),
      PropTypes.string,
    ]).description('Gap sizes between rows and/or columns.'),
    justify: PropTypes.oneOf(['start', 'center', 'end', 'stretch'])
      .description(
        `How to align the individual items inside the grid when there is extra
space in the row axis.`,
      )
      .defaultValue('stretch'),
    justifyContent: PropTypes.oneOf([
      'start',
      'center',
      'end',
      'between',
      'around',
      'stretch',
    ]).description('How to align the contents along the row axis.'),
    rows: PropTypes.oneOfType([
      PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.oneOf(sizes), PropTypes.string])),
          PropTypes.oneOf(sizes),
          PropTypes.string,
        ]),
      ),
      PropTypes.oneOf(fixedSizes),
      PropTypes.string,
    ]).description(
      `Row sizes.
      If an array value is an array, the inner array indicates the
      minimum and maximum sizes for the row.
      Specifying a single string will cause automatically added rows to be
      the specified size.`,
    ),
    tag: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
      .description(`The DOM tag to use for the element. NOTE: This is deprecated
      in favor of indicating the DOM tag via the 'as' property.`),
    as: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
      .description('The DOM tag or react component to use for the element.')
      .defaultValue('div'),
  }

  return DocumentedGrid
}

export const themeDoc = {
  'global.size': {
    description: 'The possible sizes for row and column.',
    type: 'object',
    defaultValue: `{
      xxsmall: '48px',
      xsmall: '96px',
      small: '192px',
      medium: '384px',
      large: '768px',
      xlarge: '1152px',
      xxlarge: '1536px',
      full: '100%',
    }`,
  },
  'grid.extend': {
    description: 'Any additional style for the Grid.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  ...themeDocUtils.edgeStyle('The possible sizes for the grid margin and gap.'),
}
