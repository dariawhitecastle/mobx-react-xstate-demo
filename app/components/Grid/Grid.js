/* eslint-disable react-hooks/rules-of-hooks */
/* eslint-disable react/prop-types */
/**
 * Grid
 */

import React, { useContext } from 'react'
import { Grid } from 'grommet'
import { ResponsiveContext } from 'grommet/es6/contexts'
import { isPlainObject } from 'lodash'

import { doc } from './Grid.doc'
import helpers from '../../utils/helpers'

const createGrid = ({
  a11yTitle,
  align,
  alignContent,
  alignSelf,
  areas,
  as,
  columns,
  fill,
  gap,
  gridArea,
  justify,
  justifyContent,
  margin,
  rows,
  tag,
  ...rest
}) => {
  const size = useContext(ResponsiveContext)

  return (
    <Grid
      a11yTitle={a11yTitle}
      align={align}
      alignContent={alignContent}
      alignSelf={alignSelf}
      areas={isPlainObject(areas) ? areas[size] : areas}
      as={as}
      columns={columns}
      fill={fill}
      gap={gap}
      gridArea={gridArea}
      justify={justify}
      justifyContent={justifyContent}
      margin={margin}
      rows={rows}
      tag={tag}
      {...rest}
    />
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createGrid,
})
