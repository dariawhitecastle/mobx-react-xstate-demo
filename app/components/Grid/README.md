<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createGrid
A grid system for laying out content. To use, define the
rows and columns, create area names for adjacent cells, and then
place Box components inside those areas using the gridArea property.
See https://css-tricks.com/snippets/css/complete-guide-grid/.
The availability of Grid can be tested via `Grid.available`. Use this
to create fallback rendering for older browsers, like ie11.

## Usage

```javascript
import { Grid } from 'components/Grid';
<Grid />
```

## Properties

**a11yTitle**

Custom title to be used by screen readers.

```
string
```

**alignSelf**

How to align along the cross axis when contained in
      a Box or along the column axis when contained in a Grid.

```
start
center
end
stretch
```

**gridArea**

The name of the area to place
    this inside a parent Grid.

```
string
```

**margin**

The amount of margin around the component. An object can
      be specified to distinguish horizontal margin, vertical margin, and
      margin on a particular side.

```
none
xxsmall
xsmall
small
medium
large
xlarge
{
  bottom: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  horizontal: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  left: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  right: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  top: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  vertical: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string
}
string
```

**align**

How to align the individual items inside the grid when there is extra
space in the column axis. Defaults to `stretch`.

```
start
center
end
stretch
```

**alignContent**

How to align the contents along the column axis.

```
start
center
end
between
around
stretch
```

**areas**

Area names and column,row coordinates.

```
[{
  name: string,
  start: [number],
  end: [number]
}]
```

**columns**

Column sizes.
      If an array value is an array, the inner array indicates the
      minimum and maximum sizes for the column.
      Specifying a single string will repeat multiple columns
      of that size, as long as there is room for more.
      Specifying an object allows indicating how the columns
      stretch to fit the available space.

```
[
  [
    xsmall
    small
    medium
    large
    xlarge
    full
    1/2
    1/3
    2/3
    1/4
    2/4
    3/4
    flex
    auto
    string
  ]
  xsmall
  small
  medium
  large
  xlarge
  full
  1/2
  1/3
  2/3
  1/4
  2/4
  3/4
  flex
  auto
  string
]
xsmall
small
medium
large
xlarge
{
  count: 
    fit
    fill
    number,
  size: 
    xsmall
    small
    medium
    large
    xlarge
    [xsmall
small
medium
large
xlarge
full
1/2
1/3
2/3
1/4
2/4
3/4
flex
auto]
    string
}
string
```

**fill**

Whether the width and/or height should fill the container.

```
horizontal
vertical
boolean
```

**gap**

Gap sizes between rows and/or columns.

```
small
medium
large
none
{
  row: 
    small
    medium
    large
    none
    string,
  column: 
    small
    medium
    large
    none
    string
}
string
```

**justify**

How to align the individual items inside the grid when there is extra
space in the row axis. Defaults to `stretch`.

```
start
center
end
stretch
```

**justifyContent**

How to align the contents along the row axis.

```
start
center
end
between
around
stretch
```

**rows**

Row sizes.
      If an array value is an array, the inner array indicates the
      minimum and maximum sizes for the row.
      Specifying a single string will cause automatically added rows to be
      the specified size.

```
[
  [
    xsmall
    small
    medium
    large
    xlarge
    full
    1/2
    1/3
    2/3
    1/4
    2/4
    3/4
    flex
    auto
    string
  ]
  xsmall
  small
  medium
  large
  xlarge
  full
  1/2
  1/3
  2/3
  1/4
  2/4
  3/4
  flex
  auto
  string
]
xsmall
small
medium
large
xlarge
string
```

**tag**

The DOM tag to use for the element. NOTE: This is deprecated
      in favor of indicating the DOM tag via the 'as' property.

```
string
function
```

**as**

The DOM tag or react component to use for the element. Defaults to `div`.

```
string
function
```
  
## Intrinsic element

```
div
```
## Theme
  
**global.size**

The possible sizes for row and column. Expects `object`.

Defaults to

```
{
      xxsmall: '48px',
      xsmall: '96px',
      small: '192px',
      medium: '384px',
      large: '768px',
      xlarge: '1152px',
      xxlarge: '1536px',
      full: '100%',
    }
```

**grid.extend**

Any additional style for the Grid. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.edgeSize**

The possible sizes for the grid margin and gap. Expects `object`.

Defaults to

```
{
      edgeSize: {
        none: '0px',
        hair: '1px',
        xxsmall: '3px',
        xsmall: '6px',
        small: '12px',
        medium: '24px',
        large: '48px',
        xlarge: '96px',
        responsiveBreakpoint: 'small',
      },
    }
```
