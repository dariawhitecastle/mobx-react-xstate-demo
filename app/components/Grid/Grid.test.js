import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Grommet } from 'grommet'
import Grid from './Grid'

test('Grid renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid rows renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid rows={['small', 'large', 'medium']} />
      <Grid rows="small" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid columns renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid columns={['1/2', '2/4']} />
      <Grid columns={['1/3', '2/3']} />
      <Grid columns={['1/4', '3/4']} />
      <Grid columns="small" />
      <Grid columns={{ count: 'fit', size: 'small' }} />
      <Grid columns={{ count: 'fill', size: ['small', 'medium'] }} />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid areas renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid
        rows={['xxsmall', 'medium', 'xsmall']}
        columns={['3/4', '1/4']}
        areas={[
          { name: 'header', start: [0, 0], end: [0, 1] },
          { name: 'main', start: [1, 0], end: [1, 0] },
          { name: 'sidebar', start: [1, 1], end: [1, 1] },
          { name: 'footer', start: [2, 0], end: [2, 1] },
        ]}
      />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid justify renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid justify="start" />
      <Grid justify="center" />
      <Grid justify="end" />
      <Grid justify="stretch" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid align renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid align="start" />
      <Grid align="center" />
      <Grid align="end" />
      <Grid align="stretch" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid justifyContent renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid justifyContent="start" />
      <Grid justifyContent="center" />
      <Grid justifyContent="between" />
      <Grid justifyContent="around" />
      <Grid justifyContent="end" />
      <Grid justifyContent="stretch" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid alignContent renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid alignContent="start" />
      <Grid alignContent="center" />
      <Grid alignContent="between" />
      <Grid alignContent="around" />
      <Grid alignContent="end" />
      <Grid alignContent="stretch" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid gap renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid gap="small" />
      <Grid gap="medium" />
      <Grid gap="large" />
      <Grid gap={{ row: 'small' }} />
      <Grid gap={{ row: 'medium' }} />
      <Grid gap={{ row: 'large' }} />
      <Grid gap={{ column: 'small' }} />
      <Grid gap={{ column: 'medium' }} />
      <Grid gap={{ column: 'large' }} />
      <Grid gap={{ row: 'small', column: 'medium' }} />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid fill renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid fill />
      <Grid fill={false} />
      <Grid fill="horizontal" />
      <Grid fill="vertical" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid as renders', () => {
  const component = renderer.create(
    <Grommet>
      <Grid as="article" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Grid proxies tag', () => {
  const tagComponent = renderer.create(
    <Grommet>
      <Grid tag="article" />
    </Grommet>,
  )
  const asComponent = renderer.create(
    <Grommet>
      <Grid as="article" />
    </Grommet>,
  )
  expect(tagComponent.toJSON()).toEqual(asComponent.toJSON())
})
