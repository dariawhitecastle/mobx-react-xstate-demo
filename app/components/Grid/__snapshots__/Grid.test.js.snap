// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid align renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid alignContent renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid areas renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header main footer" "header sidebar footer" ". .";
  grid-template-columns: 75% 25%;
  grid-template-rows: 48px 384px 96px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid as renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <article
    className="c1"
  />
</div>
`;

exports[`Grid columns renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 50% 50%;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 33.33% 66.66%;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 25% 75%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(192px,1fr) );
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(192px,1fr) );
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(192px,384px) );
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid fill renders 1`] = `
.c2 {
  display: grid;
  box-sizing: border-box;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  height: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid gap renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 12px 12px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 24px 24px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 48px 48px;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 12px;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 24px;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 48px;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 12px;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 24px;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 48px;
}

.c10 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 12px;
  grid-column-gap: 24px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
</div>
`;

exports[`Grid justify renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  justify-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  justify-items: stretch;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid justifyContent renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid rows renders 1`] = `
.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-rows: 192px 768px 384px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-auto-rows: 192px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
